velox/type/SimpleFunctionApi.h

```diff
@@ -235,6 +235,27 @@ struct ConstantChecker {
       isConstantType<TArgs>::value...};
 };
 
+template <typename T>
+struct ConstantArg {
```

Comment:
@PHILO-HE Thank you for working on this. Would you show an example of a how 'initialize' API would change? Also, wondering if it would make sense to use `exec::optional_arg_type<T>` instead of this new class.

CC: @kagamiori @bikramSingh91

Reply:
> Would you show an example of a how 'initialize' API would change?

@mbasmanova, I just updated this pr. You can see all `intialize` methods are affected due to arg type changing.

> wondering if it would make sense to use exec::optional_arg_type<T> instead of this new class.

Yes, this existing type is enough. Have tried using it in the latest code. Could you take a high-level review firstly?